test/e2e/tests/onboarding.spec.js

```diff
@@ -305,7 +305,10 @@ describe('MetaMask onboarding @no-mmi', function () {
           text: notificationMessage,
         });
         assert.equal(networkNotification, true);
-        await driver.clickElement({ text: 'Dismiss', tag: 'h6' });
+        await driver.clickElement({
```

Comment:
Hmm, not following why this change was needed

Reply:
the test didnt really make sense to begin with, since why would you dismiss the dialog that says 'switch to localhost', and then expect it to have switched anyways? 

Im not sure why it was passing to begin with, nor do I really know why it started failing because of this PR.

Reply:
Is it possible this wasn't a switchNetworkConfirmation previously?

Reply:
Going to look into this.

Reply:
<img width="1176" alt="Screenshot 2024-02-06 at 12 46 11 PM" src="https://github.com/MetaMask/metamask-extension/assets/34557516/5e92e8ed-165f-48ee-a557-f6dc2794be92">

so this is what things look like right before `dismiss` is clicked in the test **on develop**

Reply:
<img width="1179" alt="Screenshot 2024-02-06 at 12 54 37 PM" src="https://github.com/MetaMask/metamask-extension/assets/34557516/fc431a33-0b3b-4213-802d-2db775ce5593">

and this is what I see on this branch... clicking dismiss seems to work fine. It is weird that this switch confirmation modal is coming up when the network is already switched.

Reply:
Oh but the ETH balance is not showing up unless you click switch now... even though you appear to already be on localhost 8546

Reply:
Ok so still not sure why this PR broke this test but here is a PR against develop to fix an underlying bug with the add custom network manually flow: https://github.com/MetaMask/metamask-extension/pull/22832

cc @Gudahtt